src/infra/src/table/migration/m20250109_092400_recreate_tables_with_ksuids.rs

```diff
@@ -0,0 +1,1156 @@
+        Table::create()
+        .table(Dashboards::Table)
+        // The ID is a 27-character human readable KSUID.
+        .col(
+            ColumnDef::new(Dashboards::Id)
+                .char_len(27)
+                .not_null()
+                .primary_key(),
+        )
+        // A user-facing ID for the folder. This value can be a 64-bit signed
+        // integer "snowflake".
+        .col(ColumnDef::new(Dashboards::DashboardId).string_len(256).not_null())
+        // Foreign key to the folders table. This is a 27-character human readable
+        // KSUID.
+        .col(ColumnDef::new(Dashboards::FolderId).char_len(27).not_null())
+        // Identifier of the user that owns the dashboard.
+        .col(ColumnDef::new(Dashboards::Owner).string_len(256).not_null())
+        .col(ColumnDef::new(Dashboards::Role).string_len(256).null())
+        .col(ColumnDef::new(Dashboards::Title).string_len(256).not_null())
+        .col(ColumnDef::new(Dashboards::Description).text().null())
+        .col(ColumnDef::new(Dashboards::Data).json().not_null())
+        .col(ColumnDef::new(Dashboards::Version).integer().not_null())
+        .col(
+            ColumnDef::new(Dashboards::CreatedAt)
+                .big_integer()
+                .not_null(),
+        )
+        .foreign_key(
+            sea_query::ForeignKey::create()
+                    .name(DASHBOARDS_FOLDERS_FK)
+                    .from(Dashboards::Table, Dashboards::FolderId)
+                    .to(new_folders::Folders::Table, new_folders::Folders::Id)
+        )
+        .to_owned()
+    }
+
+    /// Statement to create unique index on dashboard_id.
+    pub fn create_dashboards_folder_id_dashboard_id_idx_stmnt() -> IndexCreateStatement {
+        sea_query::Index::create()
+            .if_not_exists()
+            .name(DASHBOARDS_FOLDER_ID_DASHBOARD_ID_IDX)
+            .table(Dashboards::Table)
+            .col(Dashboards::FolderId)
+            .col(Dashboards::DashboardId)
+            .unique()
+            .to_owned()
+    }
+
+    /// Select each record from the `legacy_dashboards` table joined on the
+    /// `legacy_folders` table and use it to create a new record in the new
+    /// `dashboards` table.
+    pub async fn populate<C: ConnectionTrait>(conn: &C) -> Result<(), DbErr> {
+        let mut legacy_dashboard_pages = legacy_entities::legacy_dashboards::Entity::find()
+            .find_also_related(legacy_entities::legacy_folders::Entity)
+            .paginate(conn, 100);
+        while let Some(legacy_dashboards) = legacy_dashboard_pages.fetch_and_next().await? {
+            let conversions_rslt: Result<Vec<_>, _> = legacy_dashboards
+                .into_iter()
+                .map(new_entities::dashboards::Model::try_from)
+                .collect();
+            let active_models = conversions_rslt?
+                .into_iter()
+                .map(|m| m.into_active_model())
+                .collect_vec();
```

Comment:
```suggestion
            let active_models = legacy_dashboards
                .into_iter()
                .map(|legacy_models| {
                    new_entities::dashboards::Model::try_from(legacy_models)
                        .map(new_entities::dashboards::Model::into_active_model)
                })
                .collect::<Result<Vec<_>, _>>()?;
```
think we can do one loop instead, considering we are migrating all three tables for this one.

Reply:
Ah sorry i missed this before merging. You're right that this would have worked